extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/context/GrpcRequestContextCdiInterceptor.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.grpc.runtime.supports.context;
+
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+
+@Interceptor
+@CleanUpRequestContext
+public class GrpcRequestContextCdiInterceptor {
+    public static final String GRPC_REQUEST_CONTEXT_STATE = "GRPC_REQUEST_CONTEXT_STATE";
+
+    @AroundInvoke
+    public Object cleanUpContext(InvocationContext invocationContext) throws Exception {
+        boolean cleanUp = false;
+        ManagedContext requestContext = Arc.container().requestContext();
+        if (!requestContext.isActive()) {
```

Comment:
@cescoffier does this look okay? I'm chaning the logic to initialize request context in a CDI interceptor. But I need to somehow pass it to the grpc interceptor `close` method to destroy the context.
I used `Vertx.currentContext()` for this and I'm accessing it [here](https://github.com/quarkusio/quarkus/pull/16830/files#diff-e06152b8deda49666ec8a46293ccb49bef940d1d3ee1833c058fd7740a71c68eR39)

Reply:
So the question is whether we can be 100% sure that the service bean is always invoked on the event loop context..

Reply:
Did you try with a method marked with `@Blocking`?

Reply:
It looks fine too me. I know there have been changed in Vert.x 4 about context management but it should be ok.

Reply:
Simply adding `@Blocking` on a method that returns multi works fine in my tests

Reply:
hmm.. But when I add Blocking and do something blocking, I'm getting an error saying that I cannot block the event loop :(

Reply:
fixed in the current version :)